src/main/scala/org/apache/spark/sql/delta/commands/ConvertToDeltaCommand.scala

```diff
@@ -199,8 +199,9 @@ abstract class ConvertToDeltaCommandBase(
     val dateFormatter = DateFormatter()
     val timestampFormatter =
       TimestampFormatter(timestampPartitionPattern, java.util.TimeZone.getDefault)
+    val dir = if (file.isDir) file.getPath else file.getPath.getParent
     val (partitionOpt, _) = PartitionUtils.parsePartition(
-      path,
```

Comment:
if path is no longer used, we do not need to have it.

Reply:
The path store the absolute path to a file. It is used later to create an AddFile action. But we don't need it to parse the partition in the directory.

Reply:
Hi @rahulsmahadev please let me know if you have other comments on this pr.